pkg/util/tolerations/tolerations.go

```diff
@@ -71,6 +73,10 @@ func MergeTolerations(first []api.Toleration, second []api.Toleration) []api.Tol
 			mergedTolerations = append(mergedTolerations, v1)
 		}
 	}
+	// order lexically by key
+	sort.Slice(mergedTolerations[:], func(i, j int) bool {
+		return mergedTolerations[i].Key < mergedTolerations[j].Key
+	})
```

Comment:
Inconsistent order here would cause the reinvocation testing to flake occasioncally.